tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/runconfig/application/BallerinaApplicationRunningState.java

```diff
@@ -67,18 +68,36 @@ public void startNotified(ProcessEvent event) {
 
     @Override
     protected BallerinaExecutor patchExecutor(@NotNull BallerinaExecutor executor) throws ExecutionException {
-        RunConfigurationKind kind = getConfiguration().getRunKind();
+        // RunConfigurationKind kind = getConfiguration().getRunKind();
         Project project = myConfiguration.getProject();
         VirtualFile baseDir = project.getBaseDir();
-        String filePath = myConfiguration.getPackage();
+        String filePath;
 
         // Find the file in the project. This is needed to find the module. Otherwise if the file is in a sub-module
         // and the SDK for the project is not set, SDK home path will be null.
         PsiFile file = BallerinaPsiImplUtil.findFileInProject(project, myConfiguration.getFilePath());
-        Module module = null;
-        if (file != null) {
-            module = ModuleUtilCore.findModuleForPsiElement(file);
+
+        assert file != null;
+        VirtualFile fileDir = file.getVirtualFile().getParent();
+        // Sets source root of the IDEA project as the termination of the recursive search.
+        String rootDir = project.getBasePath();
+        String sourcerootDir = getSourceRoot(fileDir.getPath(), rootDir);
+
+        if (sourcerootDir.equals("")) {  // if no ballerina project is found.
+            sourcerootDir = baseDir.getPath();
+            filePath = file.getVirtualFile().getPath().replace(sourcerootDir, "").substring(1);
+        } else {
+            String relativeFilePath = file.getVirtualFile().getPath().replace(sourcerootDir, "").substring(1);
+            if (relativeFilePath.contains(File.separator)) { //If file is found in a ballerina package, runs the whole
```

Comment:
Move the comment above the `if` statement.

Reply:
Done